---
int32 clock_value
